### PR TITLE
feat(developer): support comma in modifiers 🙀

### PR DESCRIPTION
**Pull request by srl295**
- improve modifier parsing
- test cases for keys and layr

Fixes: #11040

@keymanapp-test-bot skip

### CODE REVIEW

**Comment by mcdurdin** [comment]
> Fixes: feat(developer) ldml: comma in modifiers #11040

Should just be `Fixes #11040` so that GH picks up the link automatically, or `Fixes #11040 (feat(developer) ldml: comma in modifiers)` -- although I don't really like quoting the referred issue title because they tend to change over time

File: developer/src/kmc-ldml/src/util/util.ts [hunk 1]
```diff
@@ -156,14 +156,19 @@ export function verifyValidAndUnique(
...
+  return modifiers.split(',').map(m => translateModifierSubsetToLayer(m)).sort();
+}
+
+function translateModifierSubsetToLayer(modifiers: string) : number {
+  // TODO-LDML: Default #11072
   if (modifiers) {
-    // TODO-LDML
     if (modifiers.indexOf(',') !== -1) {
-      throw Error(`TODO-LDML #9838: ”,” in modifiers not supported yet.`);
+      throw Error(`This function only takes a single subset of the modifiers`);
```
Comment (mcdurdin) [inline developer/src/kmc-ldml/src/util/util.ts:171]:
```suggestion
      throw Error(`translateModifierSubsetToLayer only takes a single subset of the modifiers`);
```

**Review by mcdurdin — APPROVED** [review]

**Comment by srl295** [comment]
> > Fixes: feat(developer) ldml: comma in modifiers #11040
> 
> 
> 
> Should just be `Fixes #11040` so that GH picks up the link automatically, or `Fixes #11040 (feat(developer) ldml: comma in modifiers)` -- although I don't really like quoting the referred issue title because they tend to change over time

Ah. And here I assumed VSCode autocomplete knew something I didn't!

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.297-beta](https://keyman.com/downloads/releases/beta/17.0.297)

**Pull request merged**